File: sqry-core/fuzz/corpus/go_plugin/seeds/comprehensive.go
package main

import "fmt"

type Point struct {
    X int
    Y int
}

func (p *Point) Move(dx, dy int) {
    p.X += dx
    p.Y += dy
}

func Add(a, b int) int {
    return a + b
}

func main() {
    p := &Point{X: 0, Y: 0}
    p.Move(5, 10)
    fmt.Println(p)
}

File: sqry-core/fuzz/corpus/rust_plugin/seeds/comprehensive.rs
use std::fmt::Display;

pub struct Point {
    pub x: i32,
    pub y: i32,
}

impl Point {
    pub fn new(x: i32, y: i32) -> Self {
        Point { x, y }
    }

    pub fn distance(&self) -> f64 {
        ((self.x.pow(2) + self.y.pow(2)) as f64).sqrt()
    }
}

pub trait Drawable {
    fn draw(&self);
}

impl Drawable for Point {
    fn draw(&self) {
        println!("Point({}, {})", self.x, self.y);
    }
}

pub async fn fetch_data() -> Result<String, String> {
    Ok("data".to_string())
}
